### PR TITLE
IZE-598 added generation of markdown files with IZE specification

### PR DESCRIPTION
**Pull request by psihachina**
## Change:
added generation of markdown files with IZE specification

## Test:
[asciicast](https://asciinema.org/a/RiV0MZx8pg51Fbnh2D3Zbuty8)

### Ecs Section
| Parameter | Required | Description |
| --- | --- | --- |
| aws_profile | no | (optional) ECS-specific AWS profile (optional) can be specified here (but normally it should be inherited from a global AWS_PROFILE). |
| aws_region | no | (optional) ECS-specific AWS Region of this environment should be specified here. Normally global AWS_REGION is used. |
| cluster | no | (optional) ECS cluster can be specified here. By default it's derived from env & namespace |
| depends_on | no | (optional) expresses startup and shutdown dependencies between apps |
| docker_registry | no | (optional) Docker registry can be set here. By default it uses ECR repo with the name of the service. |
| icon | no | (optional) set icon |
| image | no | (optional) Docker image can be specified here. By default it's derived from the app name. |
| path | no | (optional) Path to ecs app folder can be specified here. By default it's derived from apps path and app name. |
| skip_deploy | no | (optional) skip deploy app. |
| task_definition_revision | no | (optional) Task definition revision can be specified here. By default latest revision is used to perform a deployment. Normally this parameter can be used via cli during specific deployment needs. |
| timeout | no | (optional) ECS deployment timeout can be specified here. |
| unsafe | no | (optional) Enables unsafe mode that increases deploy time on a cost of shorter healthcheck. |

### Main Section
| Parameter | Required | Description |
| --- | --- | --- |
| alias | no | (optional) Alias mode can be enabled here. This can be used to combine various apps via depends_on parameter. |
| apps_path | no | (optional) Path to apps directory can be set. By default apps are searched in 'apps' and 'projects' directories. This is needed in case your repo structure is not purely ize-structured (let's say you have 'src' repo in your dotnet app, as an example) |
| aws_profile | yes| (optional) AWS Profile can be specified here (but normally it's specified via AWS_PROFILE env var) |
| aws_region | yes| (required) AWS Region of this environment should be specified here. Can be overridden by AWS_PROFILE env var or --aws-region flag. |
| config_file | no | (optional) Path to ize.toml config file can be specified, but normally it's read from the environment's directory automatically. |
| custom_prompt | no | (optional) Custom prompt can be enabled here for all console connections. Default: false. |
| docker_registry | no | (optional) Docker registry can be set here. By default it uses ECR repo with the name of the service. |
| ecs | no | Ecs apps configuration. |
| env | yes| (optional) Environment name can be specified here. Normally it should be passed via `ENV` variable or --env flag. |
| env_dir | yes| (optional) Environment directory can be specified here. Normally it's calculated automatically based on the directory structure convention. |
| home | yes| (optional) User home directory can be specified here. Normally $HOME is used. |
| infra | no | Infrastructure configuration. |
| ize_dir | yes| (optional) Ize directory can be specified here. Normally it's assumed to be .infra or .ize in the current repo. |
| log_level | no | (optional) Log level can be specified here. Possible levels: info, debug, trace, panic, warn, error, fatal(default). Can be overridden via IZE_LOG_LEVEL env var or via --log-level flag. |
| namespace | yes| (required) Namespace of the project can be specified here. It is used as a base for all naming. It can be overridden by NAMESPACE env var or --namespace flag. |
| plain_text | no | (optional) Plain text output can be enabled here. Default is false. Can be overridden by IZE_PLAIN_TEXT env var or --plain-text flag. |
| prefer_runtime | yes| (optional) Prefer a specific runtime. (native or docker) (default 'native') |
| root_dir | yes| (optional) Project directory can be set here. By default it's the current directory, but in case you prefer to run ize from the outside of repo it may be useful (uncommon). |
| serverless | no | Serverless apps configuration. |
| tag | no | (optional) Tag can be set statically. Normally it is being constructed automatically based on the git revision. |
| terraform | no | Terraform configuration. |
| terraform_version | no | (optional) Terraform version can be set here. 1.1.3 by default |
| tf_log | no | (optional) Terraform TF_LOG can be set here.  Can be TRACE, DEBUG, INFO, WARN or ERROR. |
| tf_log_path | no | (optional) TF_LOG_PATH can be set here. |
| tunnel | no | Tunnel configuration. |

### CODE REVIEW

File: internal/commands/doc.go [hunk 1]
```diff
@@ -33,3 +59,42 @@ func NewCmdDoc() *cobra.Command {
...
+
+type Section struct {
+	Name string
+	schema.Items
+}
+
+var sectionTmpl = `# {{.Name}} Section
+| Parameter | Default Value | Required | Description |
+| --- | --- | --- | --- |
+{{range $k, $v := .Items}}| {{$k}} | {{$v.Default}} | {{$v.Required}} | {{$v.Description}} |
```
Comment (AutomationD) [inline internal/commands/doc.go:98]: Let's use `yes` if the parameter is required and `no` if it's not (and not true/false).

Comment (AutomationD) [reply internal/commands/doc.go:98]: If there is no way to get the default, let's remove this field from the template for now (but keep in the struct)

**Review by AutomationD — CHANGES_REQUESTED** [review]

**Review by igorkotof — APPROVED** [review]
LGTM, if @AutomationD's comments are fixed

File: internal/commands/doc.go [hunk 2]
```diff
@@ -33,3 +59,42 @@ func NewCmdDoc() *cobra.Command {
...
+
+type Section struct {
+	Name string
+	schema.Items
+}
+
+var sectionTmpl = `# {{.Name}} Section
+| Parameter | Required | Description |
+| --- | --- | --- |
+{{range $k, $v := .Items}}| {{$k}} | {{if $v.Required}}yes{{else}}no {{end}}| {{$v.Description}} |
```
Comment (AutomationD) [inline internal/commands/doc.go:98]:
```suggestion
{{range $k, $v := .Items}}| {{$k}} | {{if $v.Required}}yes{{else}}no{{end}}| {{$v.Description}} |
```

**Review by AutomationD — CHANGES_REQUESTED** [review]
Should it be like that?

**Review by AutomationD — APPROVED** [review]
Thank you @psihachina, clean code, as always.

**Review by igorkotof — APPROVED** [review]

**Pull request merged**